console/network/src/lib.rs

```diff
@@ -98,6 +98,10 @@ pub trait Network:
     const STARTING_SUPPLY: u64 = 1_500_000_000_000_000; // 1.5B credits
     /// The cost in microcredits per byte for the deployment transaction.
     const DEPLOYMENT_FEE_MULTIPLIER: u64 = 1_000; // 1 millicredit per byte
+    /// The cost in microcredits per constraint for the deployment transaction.
+    const SYNTH_FEE_MULTIPLIER: u64 = 25; // 25 microcredit per constraint to synthesize
+    /// The maximum number of constraints in a deployment
+    const MAX_DEPLOYMENT_CONSTRAINTS: u64 = 1_000_000;
```

Comment:
As the network gets better at handling large deployments, I would guess that there will be a strong push from the network governance to increase this number, and probably rather quickly. As might be the case with any limiting constants we set such as this.

If these do change in a governance decision, upgrades are going to become quite the difficulty among the validator set - thus I think it would be wise to make at the very least some kind of initial version communication methods between nodes (perhaps for the entire stack, but perhaps also for individual actions such as this) to go along with these changes, and some kind of light protocol for how to coordinate handle inter-node communication with nodes of different versions.